src/utils/utils.js

```diff
@@ -70,3 +70,31 @@ export function formatMessageForIntegration(
 
   return formatMessage({ id: messageId });
 }
+
+// deduplicateOnActiveZones was introduced as a potential fix for CUSTESC-36595. The goal of this function is to
+// deduplicate the list of zones returned by the Cloudflare API based on the status of the zone. This way, we hope to
+// guarantee that when normalizing the list of zones, only the active zone would show up in case of duplicates. For
+// instance, in the mentioned tickets, two zones were potentially returned: a purged as well as an active zone.
+export function deduplicateOnActiveZones(zones) {
+  let filteredZones = [];
+  const isActive = z => z.status === 'active';
+  const isSameZone = (z1, z2) => z1.name === z2.name;
```

Comment:
should we do a falsy check here too? `z1.name && z1.name === z2.name` ?

Reply:
The `name` field is the raw result coming from the Cloudflare API, so I assume it's properly populated to be honest. And nonetheless `undefined === 'something'` is falsy.